students/Tianx/Lesson5/mailroom3.py

```diff
@@ -0,0 +1,189 @@
+        donor_list: 2D data structure (dictionary of objects).
+
+    Returns:
+        None.
+
+    """
+    print('======= The Donor List: =======')
+    print('Donor Name:\n')
+    # Not using comprehension here.
+    for donor in donor_list:
+        print(donor)
+    print('======================================')
+
+
+def adding_donor_info(name, donation, donor_list):
+    """Adding donor info to the list.
+
+    Args:
+        name: donor name
+        donation: donation amount
+        donor_list: 2D data structure (dictionary of objects).
+
+    Returns:
+        None.
+
+    """
+    if name not in donor_list:
+        added_donor = {name: [donation]}
+        donor_list.update(added_donor)
+    else:
+        donor_list[name].insert(0, donation) # If donor name exists, insert the donation amount rather than updating the dictionary
+
+
+def send_thank_you():
+    """Sending a thank you email using user input.
+
+    Args:
+       None
+
+    Returns:
+       None
+
+    """
+    # Ask user for a donor's name or to display current list of donors, then ask for donation amount
+    print()
+    while True:
+        donor_name = input('======= The Thank you Menu: =======\n'
+                           "Enter 'list' for to see the list of donors\n"
+                           "or enter full name of donor. \n"
+                           "Enter 'exit' to return to the main menu >").title()
+        if donor_name not in dict_of_donors and donor_name != "List" and donor_name != "Exit":
+            print('\"{}\" is not a current donor, adding to the list...'.format(donor_name))
+            response = input('Please enter a donation amount for ' + donor_name + ' >')
+            while True:
+                try:
+                    float(response)
+                    break
+                except ValueError:
+                    response = input('Error: Please enter a number for the donation amount>')
+            donation_amount = response # Convert the amount into a number
+            adding_donor_info(donor_name, donation_amount, dict_of_donors)
+            create_email(donor_name, donation_amount)
+        elif donor_name == "List": #If the user types list show them a list of the donor names and re-prompt.
+            show_donor_list(dict_of_donors)
+        elif donor_name == "Exit": #If the user types exist return to main menu.
+            break
+        elif donor_name in dict_of_donors: #If user enters exsiting donor
+            response = input('Please enter a donation amount for ' + donor_name + ' >')
+            while True:
+                try:
+                    float(response)
+                    break
+                except ValueError:
+                    response = input('Error: Please enter a number for the donation amount>')
+            donation_amount = response
+            adding_donor_info(donor_name, donation_amount, dict_of_donors)
+            create_email(donor_name, donation_amount)
+            break
```

Comment:
you can simplify this logic a bit by re-ordering, this will make your code cleaner and so much easier to read:
```
if donor_name == "List":
    #
elif donor_name == "Exit":
    # 
else:
    donation_amount = input()
    if donor_name in dict_of_donors:
         # donors[name].append(donation_amount)
    else:
         # donors[name] = [donation_amount]
```